recipes/rust/build.sh

```diff
@@ -4,7 +4,19 @@
 make
 make install
 
+
+# Download a precompiled version of cargo.
+# This is needed to build cargo itself.
+if [[ $OSTYPE == darwin* ]]
+then
+    CARGO_STAGE0=https://static.rust-lang.org/dist/cargo-$PKG_VERSION-x86_64-apple-darwin.tar.gz
+else
+    CARGO_STAGE0=https://static.rust-lang.org/dist/cargo-$PKG_VERSION-x86_64-unknown-linux-gnu.tar.gz
+fi
+
+curl $CARGO_STAGE0 | tar xf cargo-0.16.0-x86_64-unknown-linux-gnu.tar.gz */cargo/bin/cargo --strip=2
+
```

Comment:
@johanneskoester I have an interest in this moving along, so I had a quick look at the failing tests on Travis and cirlceci (Appveyor seems to only have had connection problems).

While the direct error is only related to the wrong filename used (`cargo-$PKG...` instead of `rust-$PKG...`), it seems like installing cargo from the precompiled binary packages doesn't work easily. But how about using the corresponding release of cargo with the below ./configure adapted. That should also work for both linux and osx: https://github.com/rust-lang/cargo/archive/0.19.0.tar.gz